design/ceph-csi-operator.md

```diff
@@ -0,0 +1,84 @@
+---
+title: Integrate Ceph-CSI operator with Rook
+target-version: release-1.15
+---
+
+# Integrate Ceph-CSI operator with Rook
+
+## Overview
+
+To integrate the new Ceph-CSI operator with Rook and remove the existing CSI deployment related code, RBAC's and dead code in later releases. Also before moving forward please read the ceph-CSI-operator design doc [here](https://github.com/ceph/ceph-csi-operator/blob/main/docs/design/operator.md).
+
+## Rook Deploying Ceph-CSI-Operator
+
+There are two scenarios rook needs to manage while deploying of the new CSI-operator.
+
+1. Deployment in new cluster i.e [Fresh deployment](#fresh-deployment)
+    1. Manifest Deployment
+    2. Helm Deployment
+2. Deployment in upgrade cluster i.e [Upgrade clusters](#upgraded-clusters) cluster.
+    1. Manifest Upgrade
+    2. Helm Upgrade
+
+We need to have some bash script which will be called by some make command to automatically update the RBAC, CRDs files in rook repo from the ceph-CSI-operator repo.
+
+## Fresh deployment
+
+Rook will define the new CSI-operator defaults settings in [CRDs](#new-ceph-csi-operator-crds) in operator.yaml or operator chart `rook-ceph` `value.yaml`, and can be customizable by users. Rook will define the CSI operator and related CRs such as it is deployed by default and automatically with Rook.
+
+- **Manifests**:
+    - operator.yaml: Include the CSI operator deployment, and default CephCSI CRs
+    - crds.yaml: Include the CephCSI CRDs
+    - common.yaml: Include the RBAC needed for the CSI operator
+    - Consider whether to have separate manifests, particularly in the first release if the CSI operator is experimental
+- **Helm chart**: Add new settings in the operator and cluster chart and all others CRDs, RBAC will be added similarly to the manifest
+
+In fresh clusters, we will introduce a new setting in `operator.yaml` configmap `rook-ceph-operator-config` name `ROOK_USE_CSI_OPERATOR: true` and if `USE_CSI_OPERATOR:` is set to `true` it will also disable the CSI `ROOK_CSI_DISABLE_DRIVER`. Setting `ROOK_CSI_DISABLE_DRIVER: true` will skip the current CSI driver reconciliation and the settings configured in `operator.yaml` for the new CRs will be taken into consideration.
+
+From here we can proceed two ways to configure other settings and deploy the CSI-operator.
```

Comment:
This doesn't seem like two different ways to configure. The single approach is that the manifests are found in the examples folder (or a subdirectory), and if the user wants to change the defaults, they need to modify those manifests.